Various fixes for the Subsonic provider

khers opened the pull request:
Update py-opensonic to 5.2.0 for the updated song parsing (now includes disc_number).

With the new disc_number field, pass this information on to MA for better track sorting.

Finally, we have yet another attempt to fix the Navidrome multi-artist disc issue. The original hack did not include the calculated item id and overloaded all artists that needed it with UNKNOWN_ARTIST instead causing odd artists to be listed.

marcelveldt commented on music_assistant/server/providers/opensubsonic/sonic_provider.py:
```diff
@@ -100,13 +100,15 @@ async def handle_async_init(self) -> None:
             success = await self._run_async(self._conn.ping)
             if not success:
                 msg = (
-                    f"Failed to connect to {self.config.get_value(CONF_BASE_URL)}, "
+                    "Failed to connect to "
+                    f"{self.config.get_value(CONF_BASE_URL)}, "
                     "check your settings."
                 )
                 raise LoginFailed(msg)
```
why not raise AuthError or CredentialError here and catch it in one sweep below ?

khers replied:
It felt silly to raise the exception to be caught two lines later and repackaged into the LoginFailed anways, but will do

marcelveldt commented on music_assistant/server/providers/opensubsonic/sonic_provider.py:
```diff
@@ -369,7 +372,7 @@ def _parse_track(self, sonic_song: SonicSong) -> Track:
             # a Open Subsonic Song is not yet set and the implementations I have checked
             # do not contain this field. We should revisit this when the spec is finished
-            disc_number=0,
+            disc_number=sonic_song.disc_number if sonic_song.disc_number is not None else 0,
```
```suggestion
            disc_number=sonic_song.disc_number or 0,
```

marcelveldt commented on music_assistant/server/providers/opensubsonic/sonic_provider.py:
```diff
@@ -411,21 +414,23 @@ def _parse_track(self, sonic_song: SonicSong) -> Track:
                 # because it will not have an entry in the artists table so the best we can do it
                 # add a 'fake' id with the proper artist name and have get_artist() check for this
                 # id and handle it locally.
+                fake = f"{NAVI_VARIOUS_PREFIX}{sonic_song.artist}"
```
make fake_id would be a better name for this variable ?

khers committed:
Update py-opensubsonic to 5.2.1

Signed-off-by: Eric B Munson <eric@munsonfam.org>

khers committed:
Fix line lenghts and stop using f strings for logs

The new code formatter is adjusting code on save (and ruff is fighting
with it on commit). Fix line lengths to 100 characters. Stop using f
strings when logging and delgate the string formatting to the logger
so it can avoid the work when not printing.

Signed-off-by: Eric B Munson <eric@munsonfam.org>

khers committed:
Fix the NAVIDROME no artist id issue (finally) and handle disc numbers

This should finally fix the Navidrome multi-artist issue. Previously we
were creating a fake id for artists that are not in the Navidrome db but
do show up on tracks because the server does not have an id for them.
But the mapping created for that item used a generic id so users saw
the last artists added this was as on every multi-artist track.

Also, now that py-opensubsonic supports disc_numbers as an attribute of
songs, give that information to MA.

Signed-off-by: Eric B Munson <eric@munsonfam.org>

khers committed:
Merge branch 'dev' into subsonic-fixes
